build-tools/automation/azure-pipelines.yaml

```diff
@@ -1397,10 +1397,9 @@ stages:
     - ${{ if eq(variables['MicroBuildSignType'], 'Real') }}:
       - group: Publish-Build-Assets
     steps:
-    - checkout: self
-      submodules: recursive
-
-    - template: yaml-templates/use-dot-net.yaml
+    - template: yaml-templates/setup-test-environment.yaml
```

Comment:
Was this only needed to bring in the new `metadata` folder cleaning changes in xaprepare? The extra .nupkg artifact download/extraction, .vsix installation, and Android SDK dependency provisioning added by this template are likely worth skipping if possible. Maybe we could also add the `metadata` folder removal to the `DeleteExtractedWorkloadPacks` target if it's all that is needed?

Reply:
I thought one build from before `~\android-toolchain\dotnet\` was not provisioned fully, but I can't find it now.

I reverted this change, and we can see what it does.

Reply:
@pjcollins the error is:
```
 '"C:\Users\AzDevOps\android-toolchain\dotnet\dotnet"' is not recognized as an internal or external command,
```
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4927545&view=logs&j=3ce66085-846b-53dd-5323-3e00eb12b9a6&t=f1788e61-d854-556d-ca50-2c29dae4b93f&l=44

Reply:
Oh, I overlooked the usage of our android-toolchain dotnet install. I think we'll need to add that install to both the .msi and .pkg creation jobs, since there is nothing that would guarantee that the right version is already in place. Maybe it's worth adding a new  `Scenario_InstallDotNetPreview.cs` scenario to xaprepare that will only run the dotnet preview install step. Or we could continue to use `setup-test-environment.yaml` which is probably easier. I don't think it matters much either way since these jobs should be going away in the not too distant future?